### PR TITLE
[PW-2385] Add check notification event code supports hmac.

### PR DESCRIPTION
**Pull request by AlexandrosMor**
**Description**
Added the check isHmacSupportedEventCode to the library. This check returns true if the notification event code supports hmac calculation.

**Tested scenarios**
Add Unit test to test an existing event code in the list

### CODE REVIEW

File: src/Adyen/Util/HmacSignature.php [hunk 1]
```diff
@@ -103,4 +103,59 @@ public function isValidNotificationHMAC($hmacKey, $params)
...
         return $expectedSign == $merchantSign;
     }
+    /**
+     * Returns true when the event code support HMAC validation
+     *
+     * @param $response
+     */
+    public function isHmacSupportedEventCode($response)
+    {
+        $eventCodes = [
```
Comment (msilvagarcia) [inline src/Adyen/Util/HmacSignature.php:113]:
```suggestion
        $eventCodes = array(
```

File: src/Adyen/Util/HmacSignature.php [hunk 2]
```diff
@@ -103,4 +103,59 @@ public function isValidNotificationHMAC($hmacKey, $params)
...
+            "REQUEST_FOR_INFORMATION",
+            "SECOND_CHARGEBACK",
+            "SUBMIT_RECURRING",
+            "VOID_PENDING_REFUND",
+            "POSTPONED_REFUND",
+            "TECHNICAL_CANCEL",
+            "AUTHORISATION_ADJUSTMENT",
+            "CANCEL_AUTORESCUE",
+            "AUTORESCUE"
+        ];
```
Comment (msilvagarcia) [inline src/Adyen/Util/HmacSignature.php:155]:
```suggestion
        );
```

**Comment by cyattilakiss** [comment]
@AlexandrosMor Please see the code smell from sonar cloud!

File: tests/Unit/Util/HmacSignatureTest.php [hunk 3]
```diff
@@ -127,4 +127,22 @@ public function testHmacSignatureEscaping()
             $this->fail('Unexpected exception');
         }
     }
+
+    public function testisHmacSupportedEventCode()
```
Comment (Aleffio) [inline tests/Unit/Util/HmacSignatureTest.php:131]: very minor
```suggestion
    public function testIsHmacSupportedEventCode()
```

File: src/Adyen/Util/HmacSignature.php [hunk 4]
```diff
@@ -27,6 +27,7 @@
 
 class HmacSignature
 {
+    const EVENTCODE = "eventCode";
```
Comment (rikterbeek) [inline src/Adyen/Util/HmacSignature.php:30]: let's change this to EVENT_CODE

**Pull request merged**